graphs/iterative_deepening.py

```diff
@@ -0,0 +1,78 @@
+"""
+Finds paths in the following example map to illustrate Iterative Deepening Search:
+http://www.cse.unsw.edu.au/~cs3411/16s1/tut/wk03romania.gif
+"""
+
+
+def ids(graph, start, goal):
+    # dfs helper function to perform depth limited dfs
+    def dfs(graph, start, goal, depth, path):
+        """
+        if depth < 0 , it means node is not found with current depth given return -1
+        to indicate not found
+        """
+        if depth < 0:
+            return -1
+        path.append(start)
+        if start == goal:
+            return path
+        for i in graph[start]:
```

Comment:
What is `i` in here?
Please use appropriate names for the variables as mentioned in CONTRIBUTING.md